sdk/test/trace/probability_sampler_test.cc

```diff
@@ -0,0 +1,246 @@
+#include "opentelemetry/sdk/trace/probability_sampler.h"
+
+#include <gtest/gtest.h>
+
+using opentelemetry::sdk::trace::ProbabilitySampler;
+using opentelemetry::sdk::trace::Decision;
+using opentelemetry::sdk::trace::SamplingBehavior;
+
+TEST(ProbabilitySampler, ShouldSampleWithoutContext)
+{
+	ProbabilitySampler s1(0.01);
+	ProbabilitySampler s2(0.01, false, SamplingBehavior::ALL_SPANS);
+	ProbabilitySampler s3(0.01, false, SamplingBehavior::ROOT_SPANS_ONLY);
+	opentelemetry::trace::TraceId invalid_trace_id;
+
+	opentelemetry::trace::SpanKind span_kind = opentelemetry::trace::SpanKind::kInternal;
+
+	using M = std::map<std::string, int>;
+	M m1 = {{}};
+	opentelemetry::trace::KeyValueIterableView<M> view{m1};
+
+	// [DEFAULT] SamplingBehavior::DETACHED_SPANS_ONLY
+
+	auto sampling_result = s1.ShouldSample(nullptr, invalid_trace_id, "", span_kind, view);
```

Comment:
nit: consider declaring variables closer to where they're used, so only declare s1 before this line, and s2 farther down, etc.